common/changes/office-ui-fabric-react/master_2018-01-20-01-21.json

```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "SpinButton: Add getClassNAmes prop to allow complete customization of the component",
```

Comment:
rather nervous of this change. A better approach would be to avoid the getClassNames approach and use the 6.0 `getStyles` functional approach. We found that getClassNames required us to come up with every possible permutation of areas and states, whereas a function of state can produce a given set of area classes.

See the ghdocs/styling.md file for more description.

Reply:
@dzearing  this is another case where we need something that lets us completely override the styling in the 5.0 timeframe for Office Online.

Reply:
Should we just move it to getStyles in the 5.0 branch?